### DISCUSSION TITLE
Replication Factor Of Streams

### DISCUSSION THREAD

**Discussion by NikitaStriga** [rabbitmq/rabbitmq-stream-java-client#576] [General]
Hello everyone.

As per stream docs [replication-factor](https://www.rabbitmq.com/docs/streams#replication-factor) we have an opportunity to specify replication factor of the stream. Is it possible to specify it with stream java client? I haven't found the method in StreamCreator to specify it. Or mb it is a feature only for stream core project and using stream protocol streams  **have to** be replicated across all nodes?

Also if we can controll replication factor can I for example specify that I want to replicate my stream only to nodes 1,2 and ban it from replication for node 3,4?

Thanks!

**Comment by acogoluegnes** [comment]
`initial-cluster-size` is supported by the stream protocol but is not available in the stream Java client. You can create an issue for it and submit a PR, this is a trivial change.

It is not possible to "ban" a stream from replication on specific nodes. You have a level of control over the leader location (see in the stream locator) and it is possible to [rebalance replicas](https://www.rabbitmq.com/docs/streams#replica-management) from the CLI.
